Dockerfile

```diff
@@ -1,42 +1,33 @@
-FROM ubuntu:xenial
+FROM ubuntu:eoan
 
-LABEL authors='Chris Rust <chris.rust@solarwinds.com>, Dawid Śmiech <dawid.smiech@solarwinds.com>'
+LABEL authors='Dawid Śmiech <dawid.smiech@solarwinds.com>'
 
-ARG swisnap_version
-ARG swisnap_repo=swisnap
-
-USER root
 ENV DEBIAN_FRONTEND noninteractive
+ENV LC_ALL=C.UTF-8
 
-RUN \
-  apt-get update && \
-  apt-get -y install software-properties-common && \
-  LC_ALL=C.UTF-8 add-apt-repository -y ppa:rmescandon/yq && \
-  apt-get update && \
-  apt-get -y upgrade && \
-  apt-get -y install \
+ARG swisnap_repo=swisnap
+RUN apt-get update && apt-get install -y \
+    software-properties-common \
     apt-transport-https \
     ca-certificates \
     docker.io \
     curl \
-    yq
+    gnupg \
+  && add-apt-repository -y ppa:rmescandon/yq \
+  && echo "deb https://packagecloud.io/solarwinds/${swisnap_repo}/ubuntu/ eoan main" > /etc/apt/sources.list.d/swisnap.list \
+  && curl -L https://packagecloud.io/solarwinds/${swisnap_repo}/gpgkey | apt-key add - \
+  && apt-get update \
+  && apt-get install -y yq
```

Comment:
why installation of yq is moved here to a separate line?

Reply:
I was trying to see if I could isolate the number of apt-get update statements and it got left behind.  I've reverted the docker run commands.